cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -356,59 +379,251 @@ def parse_harvard_opinions(reporter, volume, make_searchable):
             cluster.save(index=False)
 
             logger.info("Adding citation for: %s", citation.base_citation())
-            Citation.objects.create(
-                volume=citation.volume,
-                reporter=citation.reporter,
-                page=citation.page,
-                type=map_reporter_db_cite_type(
-                    REPORTERS[citation.canonical_reporter][0]["cite_type"]
-                ),
-                cluster_id=cluster.id,
-            )
-            new_op_pks = []
-            for op in soup.find_all("opinion"):
-                # This code cleans author tags for processing.
-                # It is particularly useful for identifiying Per Curiam
-                for elem in [op.find("author")]:
-                    if elem is not None:
-                        [x.extract() for x in elem.find_all("page-number")]
-
-                auth = op.find("author")
-                if auth is not None:
-                    author_tag_str = titlecase(auth.text.strip(":"))
-                    author_str = titlecase(
-                        "".join(extract_judge_last_name(author_tag_str))
-                    )
-                else:
-                    author_str = ""
-                    author_tag_str = ""
-
-                per_curiam = True if author_tag_str == "Per Curiam" else False
-                # If Per Curiam is True set author string to Per Curiam
-                if per_curiam:
-                    author_str = "Per Curiam"
-
-                op_type = map_opinion_type(op.get("type"))
-                opinion_xml = str(op)
-                logger.info("Adding opinion for: %s", citation.base_citation())
-                op = Opinion(
-                    cluster_id=cluster.id,
-                    type=op_type,
-                    author_str=author_str,
-                    xml_harvard=opinion_xml,
-                    per_curiam=per_curiam,
-                    extracted_by_ocr=True,
-                )
-                # Don't index now; do so later if desired
-                op.save(index=False)
-                new_op_pks.append(op.pk)
+            add_citations(data["citations"], cluster.id)
+            new_op_pks = add_opinions(soup, cluster.id, citation)
 
         if make_searchable:
             add_items_to_solr.delay(new_op_pks, "search.Opinion")
 
         logger.info("Finished: %s", citation.base_citation())
 
 
+def add_citations(cites: List, cluster_id: int) -> None:
+    """Add citations to OpinionClusters
+
+    :param cites: Harvard Citation data
+    :param cluster_id: Cluster of found opinion in DB
+    :return:
+    """
+    for cite in cites:
+        citation = get_citations(cite["cite"])
+        if not citation:
+            continue
+
+        # Because of non-canonical reporters this code breaks for states like
```

Comment:
A note saying what will resolve the situation would be good for future-selves.

Reply:
√

Reply:
This was updated but it doesn't show outdated fwiw.